drivers/volume/portworx/portworx.go

```diff
@@ -428,6 +476,29 @@ func (p *portworx) SnapshotCreate(
 		return nil, getErrorSnapshotConditions(err), err
 	}
 
+	if snap.Metadata.Annotations != nil {
+		preSnapRule, present := snap.Metadata.Annotations[preSnapRuleAnnotationKey]
+		if present && len(preSnapRule) > 0 {
+			logrus.Infof("Running pre-snap rule: %s for snapshot: [%s] %s", preSnapRule, snap.Metadata.Namespace, snap.Metadata.Name)
+			taskID, _ := uuid.New()
+
+			podsWithBackgroundCmds, err := p.executeRuleForSnap(snap, preSnapRule, taskID.String())
+			if err != nil {
+				err = fmt.Errorf("failed to run pre-snap rule: %s due to: %v", preSnapRule, err)
```

Comment:
Need to add snap info in log. Maybe define a filter like PodLog which can be used everywhere for snap logs.

Reply:
Added SnapshotLog